aiida/work/workchain.py

```diff
@@ -156,13 +156,18 @@ def _do_step(self):
         except _PropagateReturn as exception:
             finished, result = True, exception.exit_code
         else:
-            if isinstance(stepper_result, (int, ExitCode)):
+            # Set result to None unless stepper_result was non-zero integer or ExitCode instance with non-zero status
+            if isinstance(stepper_result, int) and stepper_result != 0:
```

Comment:
Is this the right way to check also in py2/py3? (e.g. long in py2, ...)

Reply:
No clue to be honest. Surely @dev-zero would be able to tell us? :)

Reply:
if you suspect that `stepper_result` could be a long in Python 2, the future-proof ways to check are:

```python
import six

if isinstance(stepper_result, six.integer_types) and stepper_result != 0:
```

or

```python
import numbers  # Python built-in since 2.6

if isinstance(stepper_result, numbers.Integral) and stepper_result != 0:
```

but for `stepper_result` to be a long in Python 2, you either have to set it explicitly using `long(123)`, the `...L` literal suffix or that it grows bigger than `sys.maxint` (which is in Python 2: `9223372036854775807`) by itself.

Reply:
Since it can be a value that is returned by a user (return value from a `WorkChain` step function) and the spec is that it should allow positive integers greater than zero (which therefore technically should include longs), I will go with the six variant. Thanks @dev-zero